configure.ac

```diff
@@ -320,45 +320,6 @@ fi
 # features that Autoconf doesn't tell us about
 #
 
-
-#
```

Comment:
Anyone building fish on a system which doesn't support passing NULL to realpath(3) will find out the first time they try to run the program (hopefully via `make test`). That's becase fish calls it at startup to resolve the home directory path. Moving the failure to the configure step doesn't seem to add much value. But I deliberately made this a separate commit so that it can be easily reverted just in case someone tells us we need to support such an ancient system.